src/stores/AppStore.js

```diff
@@ -136,8 +136,16 @@ export default types
     userLabels: isFF(FF_DEV_1536) ? types.optional(UserLabels, { controls: {} }) : types.undefined,
   })
   .preProcessSnapshot((sn) => {
-    if (!sn.user) {
-      sn.user = window.APP_SETTINGS?.user ?? null;
+    if (isFF(FF_DEV_2887)) {
+      const currentUser = window.APP_SETTINGS?.user ?? null;
```

Comment:
user can also be in `sn.user`, so I'd put it as the first option to check

Reply:
I'll just always set sn.user to currentUser if currentUser exists, as its always correct (I had @makseq adjust this to make double sure its available with correct fields).

Reply:
But you miss situation when `user` is passed via the params to LabelStudio constructor. And LSF can be used as standalone module as well.

Reply:
https://github.com/heartexlabs/label-studio-frontend/blob/master/README.md?plain=1#L70-L74